mixer/adapter/cloudwatch/logHandler_test.go

```diff
@@ -0,0 +1,337 @@
+			},
+			"",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{
+				{Message: aws.String("testMessage2"), Timestamp: aws.Int64(1)},
+			},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{
+				{Message: aws.String("testMessage2"), Timestamp: aws.Int64(1)},
+			},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"",
+		},
+	}
+
+	for _, c := range cases {
+		h, ok := c.handler.(*handler)
+		if !ok {
+			t.Error("test case has the wrong type of handler.")
+		}
+
+		err := h.putLogEntryData(c.logEntryData, c.nextSequenceToken)
+		if len(c.expectedErrString) > 0 {
+			if err == nil {
+				t.Errorf("expected an error containing %s, got none", c.expectedErrString)
+
+			} else if !strings.Contains(strings.ToLower(err.Error()), c.expectedErrString) {
+				t.Errorf("expected an error containing \"%s\", actual error: \"%v\"", c.expectedErrString, err)
+			}
+		} else if err != nil {
+			t.Errorf("unexpected error: \"%v\"", err)
+		}
+	}
+}
+
+func TestSendLogEntriesToCloudWatch(t *testing.T) {
+	env := test.NewEnv(t)
+	cfg := &config.Params{
+		LogGroupName:  "TestLogGroup",
+		LogStreamName: "TestLogStream",
+	}
+
+	h := newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockCloudWatchLogsClient{resp: generateLogStreamOutput(), describeError: nil})
+
+	cases := []struct {
+		logEntryData                []*cloudwatchlogs.InputLogEvent
+		expectedCloudWatchCallCount int
+		Handler                     adapter.Handler
+	}{
+		{[]*cloudwatchlogs.InputLogEvent{}, 0, h},
+		{generateTestLogEntryData(1), 1, h},
+		{generateTestLogEntryData(10001), 2, h},
+		{
+			generateTestLogEntryData(0),
+			0,
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockCloudWatchLogsClient{
+				resp: generateLogStreamOutput(), describeError: errors.New("describe logstream failed"),
+			}),
+		},
+		{
+			generateTestLogEntryData(0),
+			0,
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockCloudWatchLogsClient{resp: generateNilLogStreamOutput(), describeError: nil}),
+		},
+		{
+			generateTestLogEntryData(1),
+			0,
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{resp: generateLogStreamOutput(), describeError: nil}),
+		},
+	}
+
+	for _, c := range cases {
+		h, ok := c.Handler.(*handler)
+		if !ok {
+			t.Error("test case has the wrong type of handler.")
```

Comment:
it would be nice perhaps if the cases had names or some identifier to improve debugging a failure here (same for other places this appears)

Reply:
added names to test cases.